packages/core-mobile/app/new/routes/_layout.tsx

```diff
@@ -56,32 +59,47 @@ export default function RootLayout(): JSX.Element | null {
 
   useLoadFonts()
 
+  // please be careful with the dependencies here
+  // this effect is responsible for redirecting users
+  // to either the sign up flow or the login modal
+  // we don't want to include any other dependencies here
   useEffect(() => {
     if (walletState === WalletState.NONEXISTENT) {
       if (router.canGoBack()) {
         navigation.dispatch(StackActions.popToTop())
       }
       router.replace('/signup')
     } else if (walletState === WalletState.INACTIVE) {
-      // Navigate to login modal when walletState is not active
+      // navigate to login modal when wallet is not active
       router.navigate('/loginWithPinOrBiometry')
-    } else if (walletState === WalletState.ACTIVE) {
-      if (canGoBackToWallet && router.canGoBack()) {
-        router.back()
-      } else {
-        router.replace('/portfolio')
-      }
     }
-  }, [walletState, router, canGoBackToWallet, navigation])
+  }, [walletState, router, navigation])
```

Comment:
not really satisfied with this solution since we are heavily depending on the navigation routes instead of our app state. for now this should do it but we should look into refactoring this logic.